Android View Appointments & Cancel Appointment

sandyl289 opened the pull request:
- `View Appointments` & `Cancel Appointment` pages (must be logged in as customer)
- Everything works!!

Thanks Tyler & Jacek for helping me!!!

# View Appointments
![image](https://user-images.githubusercontent.com/70775435/114954903-8361bb00-9e29-11eb-9a73-960f5a23b365.png)


# Cancel Appointments

## Attempt to cancel last appointment
![image](https://user-images.githubusercontent.com/70775435/114954953-9d9b9900-9e29-11eb-8860-12d4e49fabc0.png)
![image](https://user-images.githubusercontent.com/70775435/114954974-a5f3d400-9e29-11eb-99dc-7efa1998fd7a.png)
![image](https://user-images.githubusercontent.com/70775435/114955048-c91e8380-9e29-11eb-9397-0ef670ed6604.png)

## Attempt to cancel invalid appointment (not 7 days ahead) (works)
![image](https://user-images.githubusercontent.com/70775435/114955141-fcf9a900-9e29-11eb-885d-93bc3272883e.png)
![image](https://user-images.githubusercontent.com/70775435/114955178-0aaf2e80-9e2a-11eb-81d0-c827a0850136.png)

sandyl289 committed:
View Appointments + broken code

- It is possible to view the appointments
- I'll continue debugging cancel appointment tomorrow evening

sandyl289 committed:
correct msg for cancel within 1 week + broken code

- If try to cancel within 7 days: Correct error message

- If try to cancel future appointment: ERROR

sandyl289 committed:
Fix "onFailure() was not overridden" message + broken code

- Issue with cancelling future appointment

sandyl289 committed:
Add isWeekAhead()

- isWeekAhead() works
- Cancelling a future appointment works, but we need to refresh the list of appointments

sandyl289 committed:
Modify onSuccess

- Cancelling a future appointment works, but we need to refresh the list of appointments
- There are some issues to be fixed:
See https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/204#issuecomment-820081450

sandyl289 committed:
Refactor name View Appointment -> Cancel Appointment

sandyl289 committed:
View Appointments (View Only)

- View Appointments works correctly
https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/204#issuecomment-820706134

sandyl289 committed:
Rename some Android IDs (cancel -> view)

sandyl289 committed:
Cleanup + javadoc View Appointments

sandyl289 committed:
Attempt to redirect to 2nd cancel page (failed)

sandyl289 committed:
Another attempt

- Clicking on View Appointments will redirect to the correct page
- Still need to fix page 2 & onFailure

sandyl289 committed:
using onFailure part

sandyl289 committed:
View Appointments & Cancel Appointment works

- Thanks to Jacek's help!! Ì had to change JsonHttpResponseHandler() to TextHttpResponseHandler()
(See https://github.com/McGill-ECSE321-Winter2021/project-group-09/issues/204#issuecomment-820825987)

sandyl289 committed:
Some cleanups

sandyl289 committed:
Cleanups

- View Appointments & Cancel Appointment work!! :100:

sandyl289 committed:
Remove useless comments + spacing

T-Syme commented on RepairShop-Android/app/src/main/java/ca/mcgill/ecse321/repairshop/BaseActivity.java:
```diff
@@ -43,8 +43,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
             return true;
-        } else if (id == R.id.menuLogout) {
+        } else if (id == R.id.menuCancelAppointment) {
+            // Switch to View Appointments
```
Tiny detail (feel free to ignore): comment should be for cancelling appointment

sandyl289 replied:
Thanks! 
Done: https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/215/commits/7ba78082b3ccb785b96ac401b046347cf491a318

T-Syme reviewed (approved):
Looks good to me!

Ideally the .idea files shouldn't be there, but not a big deal! I wouldn't worry about it

sandyl289 committed:
Fix comment

https://github.com/McGill-ECSE321-Winter2021/project-group-09/pull/215/files/0a6b2ac3d5ad400d0e368022679e8edfc912daba#r614511710

sandyl289 commented:
- Remove Titles 
![image](https://user-images.githubusercontent.com/70775435/115022531-7f668500-9e8b-11eb-8934-98d3a65a43f4.png)
![image](https://user-images.githubusercontent.com/70775435/115024260-d705f000-9e8d-11eb-8df7-c579929743b1.png)
![image](https://user-images.githubusercontent.com/70775435/115022637-a624bb80-9e8b-11eb-88bf-345d3a088e69.png)
![image](https://user-images.githubusercontent.com/70775435/115024018-842c3880-9e8d-11eb-95b4-f73dc6fa47af.png)
![image](https://user-images.githubusercontent.com/70775435/115024490-25b38a00-9e8e-11eb-9219-1b6e47a76ca0.png)

![image](https://user-images.githubusercontent.com/70775435/115024594-4d0a5700-9e8e-11eb-89e6-2c89940a7581.png)

sandyl289 committed:
Adding paddings + remove titles

sandyl289 committed:
Merge remote-tracking branch 'origin/android' into android-cancelAppointment
